.github/workflows/test-code-samples.yaml

```diff
@@ -0,0 +1,31 @@
+name: Test API Docs Code Samples
+
+on:
+  push:
+    paths:
+      - 'apify-api/openapi/code_samples/**'
+  pull_request:
+    paths:
+      - 'apify-api/openapi/code_samples/**'
+
+jobs:
+  run-code-sample-tests:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Get current branch name
```

Comment:
Also the branch should be accessible under `{{ github.ref_name }}`. Then you shouldn't even need to call `actions/checkout`.